samples/algorithms/integer-factorization/FlameGraphCounter.cs

```diff
@@ -0,0 +1,98 @@
+using System.Collections.Generic;
+using System.Diagnostics;
+
+using Microsoft.Quantum.Simulation.Core;
+using Microsoft.Quantum.Simulation.Simulators.QCTraceSimulators;
+
+namespace Microsoft.Quantum.Simulation.QCTraceSimulatorRuntime {
+    public class FlameGraphCounter : IQCTraceSimulatorListener {
```

Comment:
Some comments here explaining what flamegraph.pl expects may help the reader interpret this sample; it wasn't clear to me, for instance, what the significance of `;` in call stack lines was until I looked at the flamegraph.pl docs.